rpki-validator/src/main/java/net/ripe/rpki/validator3/rrdp/RrdpServiceImpl.java

```diff
@@ -116,23 +124,16 @@ private boolean doStoreRepository(RpkiRepository rpkiRepository, RpkiRepositoryV
         if (notification.sessionId.equals(rpkiRepository.getRrdpSessionId())) {
             if (rpkiRepository.getRrdpSerial().compareTo(notification.serial) <= 0) {
                 try {
-                    final List<Delta> deltas = notification.deltas.stream().
-                            filter(d -> d.getSerial().compareTo(rpkiRepository.getRrdpSerial()) > 0).
-                            sorted(Comparator.comparing(DeltaInfo::getSerial)).
-                            parallel().
-                            map(di -> readDelta(notification, di)).
-                            collect(Collectors.toList());
-
-                    verifyDeltaSerials(deltas, notification, rpkiRepository);
-
-                    deltas.forEach(d -> {
-                        storage.readTx0(tx -> verifyDeltaIsApplicable(tx, d));
-                        storage.writeTx0(tx -> {
-                            storeDelta(tx, d, validationRun, rpkiRepository, changedObjects);
-                            tx.afterCommit(() -> rpkiRepository.setRrdpSerial(rpkiRepository.getRrdpSerial().add(BigInteger.ONE)));
-                        });
-                    });
-
+                    List<DeltaInfo> orderedDeltas = verifyDeltaSerials(notification, rpkiRepository);
+                    rrdpProcessingPool.submit(() -> orderedDeltas
```

Comment:
Just to understand better: 
".parallelStream()" will use the pool for which submit() is called?

Reply:
Yes. I couldn't find another easy way to (locally) set the level of parallelism. It would be nice to have a pool for downloads and another for heavy CPU processing. Maybe that's something for the future...